finetune/lm_base.py

```diff
@@ -45,11 +42,11 @@ class LanguageModelBase(object, metaclass=ABCMeta):
                        Providing more than `max_length` tokens to the model as input will result in truncation.
     """
 
-    def __init__(self, autosave_path=None, max_length=MAX_LENGTH, verbose=True):
+    def __init__(self, hparams=None, autosave_path=None, max_length=None, verbose=True):
+        self.hparams = hparams or get_default_hparms()
         self.autosave_path = autosave_path or tempfile.mkdtemp()
         _LOGGER.info("Writing intermediate checkpoints to {}".format(self.autosave_path))
-        self.max_length = max_length
-        self.autosave_path = autosave_path
+        self.max_length = max_length or self.hparams.MAX_LENGTH
```

Comment:
Feels redundant, there shouldn't be two correct ways to pass the same value.  I would just remove the max_length argument from top level and do self.max_length = self.hparams.MAX_LENGTH if you want to alias things

Reply:
For other endpoints it makes sense to have an override.  But for the `__init__` method I don't think it makes sense.